### PR TITLE
feat(crowdactions): Complete Details page designs and create a common navbar for the app

### PR DESCRIPTION
**Pull request by yasht01**

### CODE REVIEW

File: lib/presentation/crowd_action/pages/widgets/bottom_navbar.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    super.initState();
+    index = 0;
+  }
+
+  void onTapHandler(int index) {
+    if (index == 0) {
+      // context.router.push(HomeRoute());
+      context.router.popUntilRouteWithName('HomeRoute');
+      // context.router.navigate(HomeRoute());
+      // context.router.popUntilRoot();
```
Comment (Xazin) [inline lib/presentation/crowd_action/pages/widgets/bottom_navbar.dart:29]: Remove comments

File: lib/presentation/crowd_action/pages/widgets/in_spotlight_header.dart [hunk 2]
```diff
@@ -42,94 +42,95 @@ class _InSpotLightHeaderState extends State<InSpotLightHeader> {
...
-                  children: [
-                    const SizedBox(
-                      height: 35,
+            fetchingCrowdSpotLightActions: () {
+              return const CircularProgressIndicator(
+                color: kAccentColor,
+              );
+            },
+            spotLightCrowdActionsError: (_) {
+              // TODO - Implement screen with missing spot lights
```
Comment (Xazin) [inline lib/presentation/crowd_action/pages/widgets/in_spotlight_header.dart:51]: We need to tell this to Niklas so he knows we're missing this.

File: lib/presentation/crowd_action/widgets/bottom_navbar.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  const BottomNavbar({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: 75,
+      child: BottomNavigationBar(
+        // currentIndex: index,
```
Comment (Xazin) [inline lib/presentation/crowd_action/widgets/bottom_navbar.dart:13]: Remove??

File: lib/presentation/crowd_action/widgets/bottom_navbar.dart [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          ),
+          BottomNavigationBarItem(
+            icon: Icon(Icons.add_sharp),
+            label: '',
+          )
+        ],
+      ),
+    );
+  }
+}
```
Comment (Xazin) [inline lib/presentation/crowd_action/widgets/bottom_navbar.dart:37]: Newline

**Review by Xazin — CHANGES_REQUESTED** [review]

**Comment by Xazin** [comment]
https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit
These commit titles need to be clearer and more consistent.

`feat: change details page and add common appbar` etc.

File: lib/presentation/crowd_action/pages/widgets/in_spotlight_header.dart [hunk 5]
```diff
@@ -42,94 +42,95 @@ class _InSpotLightHeaderState extends State<InSpotLightHeader> {
...
+                  ),
+                  const SizedBox(
+                    height: 10,
+                  ),
+                  ExpandablePageView.builder(
+                    itemBuilder: (ctx, index) {
+                      final crowdAction = _pages[index];
+                      return CrowdActionCard(
+                        title: crowdAction.name,
+                        imagePath: crowdAction.image ?? "",
```
[the remark this answers is not in the document]
Comment (yasht01) [reply lib/presentation/crowd_action/pages/widgets/in_spotlight_header.dart:78]: @Xazin Do I add a placeholder image as a fallback path if its null?

**Comment by yasht01** [comment]
Just a few more pointers:
- The `bottom_navbar.dart` and `crowdaction_details_page_old.dart` are not required now.
- The code for bottom navbar is now placed in `home_screen.dart` file to implement nested routing with a bottom navbar.

File: lib/presentation/crowd_action/pages/widgets/in_spotlight_header.dart (hunk 2 above)
Comment (yasht01) [reply lib/presentation/crowd_action/pages/widgets/in_spotlight_header.dart:51]: This issue was raised by Isaac in yesterday's Design meet. Niklas has added this to his Todo.

File: lib/presentation/crowd_action/pages/in_spot_light.dart [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import 'package:collaction_app/application/crowdaction/spotlight/spot_light_bloc.dart';
 import 'package:collaction_app/infrastructure/core/injection.dart';
 import 'package:collaction_app/presentation/crowd_action/pages/widgets/share_collaction_card.dart';
-import 'package:collaction_app/presentation/home/widgets/current_upcoming_ca_layout.dart';
+import 'package:collaction_app/presentation/home/widgets/current_upcoming_layout.dart';
```
Comment (wizlif) [inline lib/presentation/crowd_action/pages/in_spot_light.dart:4]: Use relative imports for all **package:collaction_app** imports

File: lib/presentation/crowd_action/pages/in_spot_light.dart [hunk 7]
```diff
@@ -23,7 +23,7 @@ class InSpotLightPage extends StatelessWidget {
               InSpotLightHeader(),
               CurrentAndUpcomingLayout(),
               ShareCollActionCard(),
-              CurrentAndUpcomingLayout(isCurrent: false)
+              // CurrentAndUpcomingLayout(isCurrent: false)
```
Comment (wizlif) [inline lib/presentation/crowd_action/pages/in_spot_light.dart:26]: Remove Comment

**Review by wizlif — CHANGES_REQUESTED** [review]

File: lib/presentation/crowd_action/pages/in_spot_light.dart (hunk 6 above)
Comment (yasht01) [reply lib/presentation/crowd_action/pages/in_spot_light.dart:4]: @wizlif should I replace with relative imports for all 4 lines?

**Pull request merged**